Unable to build parsec 0.7.2 with rust 1.43.1. Parsec 0.6.0 builds fine.

Unable to build parsec 0.7.2 with rust 1.43.1. Parsec 0.6.0 builds fine.

The error is:
```
[  678s]    Compiling cryptoki v0.1.0
[  692s] error[E0277]: the trait bound `std::vec::Vec<u8>: std::convert::From<[u8; 4]>` is not satisfied
[  692s]   --> src/providers/pkcs11/key_management.rs:30:47
[  692s]    |
[  692s] 30 |         let mut template = vec![Attribute::Id(key_id.to_be_bytes().into())];
[  692s]    |                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `std::convert::From<[u8; 4]>` is not implemented for `std::vec::Vec<u8>`
[  692s]    |
[  692s]    = help: the following implementations were found:
[  692s]              <std::vec::Vec<T> as std::convert::From<&[T]>>
[  692s]              <std::vec::Vec<T> as std::convert::From<&mut [T]>>
[  692s]              <std::vec::Vec<T> as std::convert::From<std::borrow::Cow<'a, [T]>>>
[  692s]              <std::vec::Vec<T> as std::convert::From<std::boxed::Box<[T]>>>
[  692s]            and 5 others
[  692s]    = note: required because of the requirements on the impl of `std::convert::Into<std::vec::Vec<u8>>` for `[u8; 4]`
[  692s] 
[  692s] error[E0277]: the trait bound `std::vec::Vec<u8>: std::convert::From<[u8; 4]>` is not satisfied
[  692s]    --> src/providers/pkcs11/key_management.rs:106:27
[  692s]     |
[  692s] 106 |             Attribute::Id(key_id.to_be_bytes().into()),
[  692s]     |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `std::convert::From<[u8; 4]>` is not implemented for `std::vec::Vec<u8>`
[  692s]     |
[  692s]     = help: the following implementations were found:
[  692s]               <std::vec::Vec<T> as std::convert::From<&[T]>>
[  692s]               <std::vec::Vec<T> as std::convert::From<&mut [T]>>
[  692s]               <std::vec::Vec<T> as std::convert::From<std::borrow::Cow<'a, [T]>>>
[  692s]               <std::vec::Vec<T> as std::convert::From<std::boxed::Box<[T]>>>
[  692s]             and 5 others
[  692s]     = note: required because of the requirements on the impl of `std::convert::Into<std::vec::Vec<u8>>` for `[u8; 4]`
[  692s] 
[  692s] error[E0277]: the trait bound `std::vec::Vec<u8>: std::convert::From<[u8; 3]>` is not satisfied
[  692s]    --> src/providers/pkcs11/key_management.rs:125:61
[  692s]     |
[  692s] 125 |                 pub_template.push(Attribute::PublicExponent(utils::PUBLIC_EXPONENT.into()));
[  692s]     |                                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `std::convert::From<[u8; 3]>` is not implemented for `std::vec::Vec<u8>`
[  692s]     |
[  692s]     = help: the following implementations were found:
[  692s]               <std::vec::Vec<T> as std::convert::From<&[T]>>
[  692s]               <std::vec::Vec<T> as std::convert::From<&mut [T]>>
[  692s]               <std::vec::Vec<T> as std::convert::From<std::borrow::Cow<'a, [T]>>>
[  692s]               <std::vec::Vec<T> as std::convert::From<std::boxed::Box<[T]>>>
[  692s]             and 5 others
[  692s]     = note: required because of the requirements on the impl of `std::convert::Into<std::vec::Vec<u8>>` for `[u8; 3]`
[  692s] 
[  692s] error[E0277]: the trait bound `std::vec::Vec<u8>: std::convert::From<[u8; 4]>` is not satisfied
[  692s]    --> src/providers/pkcs11/key_management.rs:228:37
[  692s]     |
[  692s] 228 |         template.push(Attribute::Id(key_id.to_be_bytes().into()));
[  692s]     |                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `std::convert::From<[u8; 4]>` is not implemented for `std::vec::Vec<u8>`
[  692s]     |
[  692s]     = help: the following implementations were found:
[  692s]               <std::vec::Vec<T> as std::convert::From<&[T]>>
[  692s]               <std::vec::Vec<T> as std::convert::From<&mut [T]>>
[  692s]               <std::vec::Vec<T> as std::convert::From<std::borrow::Cow<'a, [T]>>>
[  692s]               <std::vec::Vec<T> as std::convert::From<std::boxed::Box<[T]>>>
[  692s]             and 5 others
[  692s]     = note: required because of the requirements on the impl of `std::convert::Into<std::vec::Vec<u8>>` for `[u8; 4]`
[  692s] 
[  693s] error: aborting due to 4 previous errors
```

--- a/src/providers/pkcs11/key_management.rs
+++ b/src/providers/pkcs11/key_management.rs
@@ -27,7 +27,7 @@ impl Provider {
         key_id: u32,
         key_type: KeyPairType,
     ) -> Result<ObjectHandle> {
-        let mut template = vec![Attribute::Id(key_id.to_be_bytes().into())];
+        let mut template = vec![Attribute::Id(key_id.to_be_bytes().to_vec())];
 
         match key_type {
             KeyPairType::PublicKey => template.push(Attribute::Class(ObjectClass::PUBLIC_KEY)),
@@ -103,7 +103,7 @@ impl Provider {
         let key_id = self.create_key_id();
 
         let mut pub_template = vec![
-            Attribute::Id(key_id.to_be_bytes().into()),
+            Attribute::Id(key_id.to_be_bytes().to_vec()),
             Attribute::Token(true.into()),
             Attribute::AllowedMechanisms(vec![Mechanism::try_from(
                 key_attributes.policy.permitted_algorithms,
@@ -122,7 +122,7 @@ impl Provider {
         let mech = match key_attributes.key_type {
             Type::RsaKeyPair => {
                 pub_template.push(Attribute::Private(false.into()));
-                pub_template.push(Attribute::PublicExponent(utils::PUBLIC_EXPONENT.into()));
+                pub_template.push(Attribute::PublicExponent(utils::PUBLIC_EXPONENT.to_vec()));
                 pub_template.push(Attribute::ModulusBits(
                     key_attributes.bits.try_into().map_err(to_response_status)?,
                 ));
@@ -225,7 +225,7 @@ impl Provider {
         template.push(Attribute::PublicExponent(exponent_object.into()));
         template.push(Attribute::Verify(true.into()));
         template.push(Attribute::Encrypt(true.into()));
-        template.push(Attribute::Id(key_id.to_be_bytes().into()));
+        template.push(Attribute::Id(key_id.to_be_bytes().to_vec()));
         template.push(Attribute::Private(false.into()));
         template.push(Attribute::AllowedMechanisms(vec![MechanismType::RSA_PKCS]));
 

